src/product/controllers/product-v1.controller.ts

```diff
@@ -29,4 +31,14 @@ export class ProductController {
       data: await this.productService.getProducts(),
     };
   }
+
+  @ApiOkResponse({type: ProductResponseDto})
+  @Delete()
+  async deleteProduct(@Query('productID') productID:string): Promise<DeleteProductResponse> {
```

Comment:
1. In delete cases, normally we have to delete one entity by id (this id is selected by one user in the interface sou, we could have only param: delete by id and the path is: [PATCH] v1/products/:id

2. The Http Method must be Patch because you are update one property of the entity product (soft_delete)